presto-main/src/test/java/com/facebook/presto/execution/TestSetPathTask.java

```diff
@@ -0,0 +1,130 @@
+import com.facebook.presto.sql.tree.SetPath;
+import com.facebook.presto.transaction.TransactionManager;
+import com.facebook.presto.type.TypeRegistry;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.transaction.TransactionManager.createTestTransactionManager;
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class TestSetPathTask
+{
+    private final TransactionManager transactionManager;
+    private final AccessControl accessControl;
+    private final MetadataManager metadata;
+
+    private final ExecutorService executor = newCachedThreadPool(daemonThreadsNamed("stage-executor-%s"));
+
+    public TestSetPathTask()
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        transactionManager = createTestTransactionManager(catalogManager);
+        accessControl = new AllowAllAccessControl();
+
+        metadata = new MetadataManager(
+                new FeaturesConfig(),
+                new TypeRegistry(),
+                new BlockEncodingManager(new TypeRegistry()),
+                new SessionPropertyManager(),
+                new SchemaPropertyManager(),
+                new TablePropertyManager(),
+                transactionManager);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        executor.shutdownNow();
+    }
+
+    @Test
+    public void testSetPath()
+    {
+        PathSpecification pathSpecification = new PathSpecification(
+                Optional.empty(), ImmutableList.of(
+                        new PathElement(Optional.empty(), new Identifier("foo"))));
+
+        QueryStateMachine stateMachine = getStateMachine("SET PATH foo");
+        executeSetPathTask(pathSpecification, stateMachine);
+
+        assertEquals(stateMachine.getSetPath(), "foo");
+    }
+
+    @Test (expectedExceptions = PrestoException.class)
+    public void testSetPathInvalidCatalog()
+    {
+        PathSpecification invalidPathSpecification = new PathSpecification(
+                Optional.empty(), ImmutableList.of(
+                        new PathElement(Optional.of(new Identifier("invalidCatalog")), new Identifier("thisDoesNotMatter"))));
+
+        QueryStateMachine stateMachine = getStateMachine("SET PATH invalidCatalog.thisDoesNotMatter");
+        executeSetPathTask(invalidPathSpecification, stateMachine);
+        //error should have been thrown due to nonexistent catalog name
+        fail();
```

Comment:
This `fail` call and the comment above it are not needed, because if this test doesn't throw a `PrestoException` it will be marked as failed by testng.